test/transparentproxy/install/install.go

```diff
@@ -0,0 +1,396 @@
+
+			container, err := testcontainers.GenericContainer(
+				context.Background(),
+				testcontainers.GenericContainerRequest{
+					ContainerRequest: testcontainers.ContainerRequest{
+						Image:      tc.image,
+						Privileged: true,
+						Files: []testcontainers.ContainerFile{{
+							HostFilePath:      TProxyConfig.KumactlLinuxBin,
+							ContainerFilePath: "/usr/local/bin/kumactl",
+							FileMode:          0o700,
+						}},
+						Cmd: []string{"sleep", "infinity"},
+						LifecycleHooks: []testcontainers.ContainerLifecycleHooks{
+							{PostStarts: utils.BuildContainerHooks(tc.postStart)},
+						},
+						WaitingFor: wait.ForExec([]string{"kumactl", "version"}).
+							WithStartupTimeout(time.Second * 10).
+							WithExitCodeMatcher(func(exitCode int) bool {
+								return exitCode == 0
+							}).
+							WithResponseMatcher(func(body io.Reader) bool {
+								data, _ := io.ReadAll(body)
+								return bytes.Contains(data, []byte("Client: "))
+							}),
+					},
+					Started: true,
+				},
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			DeferCleanup(func() {
+				Expect(container.Terminate(context.Background())).To(Succeed())
+			})
+
+			EnsureInstallSuccessful(container, tc.additionalFlags)
+			EnsureGoldenFiles(container, tc)
+		},
+		EntriesForImages(TProxyConfig.DockerImagesToTest),
+		XEntriesForImages(TProxyConfig.DockerImagesToTestPaused),
+	)
+}
+
+func EnsureInstallSuccessful(container testcontainers.Container, flags []string) {
+	GinkgoHelper()
+
+	exitCode, reader, err := container.Exec(
+		context.Background(),
+		append(
+			[]string{
+				"kumactl",
+				"install",
+				"transparent-proxy",
+				"--kuma-dp-user",
+				"kuma-dp",
+			},
+			flags...,
+		),
+		exec.Multiplexed(),
+	)
+	Expect(err).NotTo(HaveOccurred())
+
+	if exitCode != 0 {
+		buf := new(strings.Builder)
+		_, _ = io.Copy(buf, reader)
+
+		Fail(fmt.Sprintf("installation ended with code %d: %s", exitCode, buf))
+	}
+}
+
+func EnsureGoldenFiles(container testcontainers.Container, tc testCase) {
+	GinkgoHelper()
+
+	var errs []error
+	outputMap := map[string][]string{}
+
+	saveCmds := map[string][]string{
+		"ipv4": {
+			"iptables-save",
+			"iptables-legacy-save",
+			"iptables-nft-save",
+		},
+	}
+
+	if TProxyConfig.IPV6 {
+		saveCmds["ipv6"] = []string{
+			"ip6tables-save",
+			"ip6tables-legacy-save",
+			"ip6tables-nft-save",
+		}
+	}
+
+	for suffix, cmds := range saveCmds {
+		for _, cmd := range cmds {
+			exitCode, reader, err := container.Exec(
+				context.Background(),
+				[]string{cmd},
+				exec.Multiplexed(),
+			)
+			if err != nil {
+				errs = append(errs, errors.Wrapf(err, "%s failed", cmd))
+				continue
+			}
+
+			buf := new(strings.Builder)
+			if _, err := io.Copy(buf, reader); err != nil {
+				errs = append(errs, errors.Wrapf(
+					err,
+					"%s: copying output reader to buffer failed",
+					cmd,
+				))
+				continue
+			}
+
+			outputMap[suffix] = append(
+				outputMap[suffix],
+				fmt.Sprintf("# %s", cmd),
```

Comment:
why do we put result of all running all executables into a single golden file? Wouldn't it be easier to have them separate, like: `ubuntu-22-04-ipv4.iptables.golden`, `ubuntu-22-04-ipv4.iptables-legacy.golden`, `ubuntu-22-04-ipv4.iptables-nft.golden`. In my opinion it will be easier to immediately spot which one failed

Reply:
I did it this way as there is a huge amount of files generated when putting it in separate files, but now when I think about it I think you are right and there are more benefits than disadvantages doing it like that

Reply:
Addressed